### ISSUE TITLE
Generator is considered awaitable

### ISSUE THREAD

**Issue by matejcik** [microsoft/pyright#2847]
this is with pyright **1.1.206**

The following will typecheck:
```python
from typing import Generator

def generator() -> Generator[int, None, int]:
    yield 1
    return 1

async def awaitable() -> int:
    print(await generator())
    return 1

aw = awaitable()
aw.send(None)
```
It will very obviously fail at run-time:
```
Traceback (most recent call last):
  File "test.py", line 12, in <module>
    aw.send(None)
  File "test.py", line 8, in awaitable
    print(await generator())
TypeError: object generator can't be used in 'await' expression
```

**mypy** will correctly report the problem:
```
test.py:8: error: Incompatible types in "await" (actual type "Generator[int, None, int]", expected type "Awaitable[Any]")
```

### Further notes

As it happens, I am **very happy** with this behavior! :) I'm working in MicroPython where async/await is just syntax sugar for generators and `yield from` statements, so the above code actually works exactly as written.

Whatever the resolution of this bug, I would like to be able to trigger the behavior intentionally, e.g., by modifying a local copy of typeshed so that `Awaitable[T]` is a superclass of `Generator[X, Y, T]`

**Comment by erictraut** [comment]
Thanks for the bug report. Based on my misreading of PEP 491, I special-cased return types of `Generator`. I've removed this special casing, so pyright now correctly flags this as an error. It now requires that the return type is an object that supports the `__await__` and `__iter__` magic methods.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.207, which I just published. It will also be included in the next release of pylance.
